### PR TITLE
feat(custom-measurements): Add units to meta

### PR DESCRIPTION
**Pull request by wmak**
- This adds a new units field to our meta that compliments the `fields` dictionary so that we can have units perf field as well
- Units of "none" means that there isn't an applicable one eg. for `date`

### CODE REVIEW

File: src/sentry/search/events/constants.py [hunk 1]
```diff
@@ -86,6 +86,20 @@ class ThresholdDict(TypedDict):
 DURATION_PATTERN = re.compile(r"(\d+\.?\d?)(\D{1,3})")
 
 RESULT_TYPES = {"duration", "string", "number", "integer", "percentage", "date"}
+SIZE_UNITS = {"bit", "byte", "kibibyte", "mebibyte", "gibibyte", "tebibyte", "pebibyte", "exbibyte"}
+DURATION_UNITS = {
+    "nanosecond",
+    "microsecond",
+    "millisecond",
+    "second",
+    "second",
```
Comment (edwardgou-sentry) [inline src/sentry/search/events/constants.py:95]:
```suggestion
    "second",
```

**Review by edwardgou-sentry — APPROVED** [review]
just nits. Do we have to update the type for `MetricMeta.unit`? Right now it's just `MetricUnit = Literal["seconds"]`

File: src/sentry/search/events/constants.py (hunk 1 above)
Comment (wmak) [reply src/sentry/search/events/constants.py:95]: what it's a set though 🙈 (thanks heh)

File: src/sentry/search/events/datasets/metrics.py [hunk 2]
```diff
@@ -150,6 +159,7 @@ def function_converter(self) -> Mapping[str, fields.MetricsFunction]:
                     snql_distribution=lambda args, alias: self._resolve_percentile(
                         args, alias, 0.5
                     ),
+                    result_type_fn=self.reflective_metric_type(),
```
Comment (andriisoldatenko) [inline src/sentry/search/events/datasets/metrics.py:162]: nit: argument ends with `_fn` but we pass value, not a function.

Comment (wmak) [reply src/sentry/search/events/datasets/metrics.py:162]: Sorry, not sure what you mean, `reflective_metric_type` returns a callable?

Comment (andriisoldatenko) [reply src/sentry/search/events/datasets/metrics.py:162]: so i expect here -> `result_type_fn=self.reflective_metric_type`

Comment (andriisoldatenko) [reply src/sentry/search/events/datasets/metrics.py:162]: never mind, i miss 1 line.

**Review by andriisoldatenko — APPROVED** [review]

**Pull request merged**